src/components/CustomFieldForm.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState } from 'react';
+import { initialValuesFormik, FormValues, yupSchemaCustomForm } from '../utils/yupSchema';
+import { useFormik } from 'formik';
+import { Box, Button } from '@chakra-ui/react';
+import { sendObj } from '../utils/fetchSupabase';
+import { SingleInputField } from './SingleInputField';
+import { createObject } from '../utils/functions';
+import { SingleSelectInput } from './SingleSelectInput';
+import { dishType } from '../utils/data';
+
+// (fields depending on the dish type should be required conditionally based on what type of dish is selected).
+// ma to tutaj sens jak ustawilem wartosci startowe w initialState?? to tez ulatwia walidacje i jupscheme
+
+// w teorii jak takie warunkowe wyświetlanie powinno wyglądać, gdyby initialState były puste i required we wszystkich oknach?
```

Comment:
znaczy tutaj bazujesz zawsze na stanie formularza, jak klient zaznaczy pole które się zmieni w initial state to wyświetlasz nowe pola. Problem tylko w walidacji potem bo trzeba ją dobrze rozpisać